packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -44,29 +58,35 @@ export class DataSourceWithBackend<
         throw new Error('Unknown Datasource: ' + q.datasource);
       }
       return {
-        ...this.applyTemplateVariables(q),
         datasourceId: ds.id,
+        datasourceName: q.datasource,
+        refId: q.refId,
+        from: rangeFrom,
+        to: rangeTo,
         intervalMs,
         maxDataPoints,
-        orgId,
+        model: {
+          ...this.applyTemplateVariables(q),
+          datasourceId: ds.id,
+        },
       };
     });
 
     const body: any = {
-      expressionCount,
       queries,
     };
-    if (range) {
-      body.range = range;
-      body.from = range.from.valueOf().toString();
-      body.to = range.to.valueOf().toString();
-    }
 
     const req: Promise<DataQueryResponse> = getBackendSrv()
-      .post('/api/ds/query', body)
+      .datasourceRequest({
+        url: expressionCount > 0 ? '/api/ds/transform' : '/api/ds/query',
+        method: 'POST',
+        data: body,
+        requestId,
+      })
       .then((rsp: any) => {
-        return this.toDataQueryResponse(rsp);
+        return this.toDataQueryResponse(rsp?.data);
```

Comment:
you sure?  I thought `toDataQueryResponse` takes the raw response and decides what to do

Reply:
I tested with only rsp and it didn't work. But I may be wrong? If you have time please have a look at it.